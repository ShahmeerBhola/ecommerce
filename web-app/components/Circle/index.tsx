import React from 'react';
import Text from '../Text';

type Props = {
  className?: string;
  height?: 'h-4';
  width?: 'w-4';
  fontSize?: Styles.FontSizes;
};

const Circle: React.FunctionComponent<Props> = ({
  children,
  className,
  height = 'h-4',
  width = 'w-4',
  fontSize = 'text-xs',
}) => {
  const baseClasses = `rounded-full ${height} ${width} bg-red-200 text-white flex items-center justify-center`;
  const classes = `${baseClasses} ${className}`;
  return (
    <div className={classes}>
      <Text size={fontSize} color="text-white">
        {children}
      </Text>
    </div>
  );
};

export default Circle;
