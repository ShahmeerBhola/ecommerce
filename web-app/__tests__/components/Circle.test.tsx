import React from 'react';
import renderer from 'react-test-renderer';
import Circle from '../../components/Circle';

describe('Circle component', () => {
  test('it renders properly', () => {
    const tree = renderer
      .create(
        <Circle>
          <div>children</div>
        </Circle>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  test('it renders properly with non-default props', () => {
    const tree = renderer
      .create(
        <Circle size="60" fontSize="3xl" className="hello-there">
          <p>Hello</p>
        </Circle>,
      )
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
