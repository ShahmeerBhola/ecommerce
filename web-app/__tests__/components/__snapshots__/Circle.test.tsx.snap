// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Circle component it renders properly 1`] = `
<div
  className="rounded-full h-4 w-4 bg-red-200 text-white flex items-center justify-center"
>
  <p
    className="font-secondary font-normal text-white text-xs"
  >
    <div>
      children
    </div>
  </p>
</div>
`;

exports[`Circle component it renders properly with non-default props 1`] = `
<div
  className="rounded-full h-60 w-60 bg-red-200 text-white flex items-center justify-center hello-there"
>
  <p
    className="font-secondary font-normal text-white text-3xl"
  >
    <p>
      Hello
    </p>
  </p>
</div>
`;
